sdk/containerregistry/azure-containerregistry/azure/containerregistry/_base_client.py

```diff
@@ -44,12 +45,12 @@ def __init__(self, endpoint: str, credential: Optional[TokenCredential], **kwarg
             **kwargs
         )
 
-    def __enter__(self):
+    def __enter__(self) -> Self:
```

Comment:
Last time I tried `Self`even from typing_extensions would fail Python 3.7

There are workaround in https://peps.python.org/pep-0673/

Reply:
Updated